sdl-core/src/test/scala/io/smartdatalake/workflow/action/spark/transformer/SparkFlattenDFTransformerTest.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2023 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.action.spark.transformer
+
+import io.smartdatalake.config.SdlConfigObject.DataObjectId
+import io.smartdatalake.config.InstanceRegistry
+import io.smartdatalake.testutils.TestUtil
+import io.smartdatalake.workflow.ActionPipelineContext
+import io.smartdatalake.config.SdlConfigObject.ActionId
+import org.apache.spark.sql.{SparkSession, Row}
+import org.scalatest.FunSuite
+import org.apache.spark.sql.types.{StructType, IntegerType, StringType, ArrayType}
+import java.util.ArrayList
+
+
+
+class SparkFlattenDFTransformerTest extends FunSuite {
+
+  protected implicit val session: SparkSession = TestUtil.session
+  import session.implicits._
+
+  implicit val instanceRegistry = new InstanceRegistry()
+  implicit val context: ActionPipelineContext = TestUtil.getDefaultActionPipelineContext
+
+  test("Nested DataFrame returns a flat Dataframe") {
+    val flattenDfTransformer = SparkFlattenDfTransformer()
+    val nestedSchema: StructType = new StructType()
+      .add("name", StringType, true)
+      .add("info", new StructType()
+        .add("age", IntegerType, true)
+        .add("address", StringType, true), true)
+
+    //createDataFrame requires Java Lists
+    val nestedData = new ArrayList[Row]()
+    nestedData.add(Row("Michael", Row(30, "123 Main St")))
+    nestedData.add(Row("Bob", Row(25, "456 Elm St")))
+
+    val nested_df = session.createDataFrame(nestedData, nestedSchema)
+    val new_df = flattenDfTransformer.transform(ActionId("ActionId"), Seq(), nested_df, DataObjectId("dataObjectId"))
+    val expectedSchema = new StructType()
+      .add("name", StringType, true)
+      .add("info_age", IntegerType, true)
+      .add("info_address", StringType, true)
+    assert(new_df.schema.equals(expectedSchema))
+  }
```

Comment:
I suggest testing 2 hierarchy levels:
```suggestion
  test("Nested DataFrame returns a flat Dataframe") {
    val flattenDfTransformer = SparkFlattenDfTransformer()
    val nestedSchema: StructType = new StructType()
      .add("name", StringType, true)
      .add("info", new StructType()
        .add("age", IntegerType, true)
        .add("address", new StructType()
          .add("street", StringType)
          .add("number", StringType)
        ))

    //createDataFrame requires Java Lists
    val nestedData = new ArrayList[Row]()
    nestedData.add(Row("Michael", Row(30, Row("Main St","123"))))
    nestedData.add(Row("Bob", Row(25, Row("Elm St","456"))))

    val nested_df = session.createDataFrame(nestedData, nestedSchema)
    val new_df = flattenDfTransformer.transform(ActionId("ActionId"), Seq(), nested_df, DataObjectId("dataObjectId"))
    val expectedSchema = new StructType()
      .add("name", StringType, true)
      .add("info_age", IntegerType, true)
      .add("info_address_street", StringType)
      .add("info_address_number", StringType)
    assert(new_df.schema.equals(expectedSchema))
  }
```